Skip failing tests on xpu with complex dtype on windows

tbohutyn opened the pull request:
Fixes https://github.com/intel/torch-xpu-ops/issues/1195

On xpu we use std:: implemetation of trig kernels. Issue comes from differences in implementation of trigonometry functions on complex dtypes in compiler headers. Windows compiler implementation is not conformant with ISO 9899. For example, following code
```
#include <cmath>
#include <complex>
#include <iostream>
#include <limits>
int main() {
  std::complex<float> x(std::numeric_limits<float>::infinity(),
                        std::numeric_limits<float>::infinity());
  std::cout << std::sinh(x) << std::endl;
}

```
Compiled with g++:
`(inf,-nan)`

Compiled with msvc:
`(inf,inf)`

While ISO 9899 clearly says:

> csinh(+∞ + i∞) returns ±∞ + iNaN (where the sign of the real part of the result is unspecified) and raises the ‘‘invalid’’ floating-point exception

These tests use numpy as reference and numpy is implemented according to ISO 9899, hence those tests fail on Windows.
Same failures can be observed on cpu, and those tests are skipped there. I propose we do the same for xpu. 
(https://github.com/intel/torch-xpu-ops/issues/1195#issuecomment-3377165435)

guangyey reviewed (commented):
It would be better if we could do this design, I don't know if it is easy.
```python
DecorateInfo(unittest.expectedFailure, 'TestUnaryUfuncs', 'test_reference_numerics_large',
                                    device_type=['cuda', "xpu"],
                                    dtypes=(torch.chalf,), active_if=IS_WINDOWS),
```

EikanWang reviewed (dismissed):
Compared to CUDA, the major change in this PR is device_type, and it is safe to assume that CUDA and XPU are exclusive within a single machine. So, I'm considering whether we should provide a variable like `GPU_TYPE` as the same as the practice in Inductor test cases. 

@albanD , any comments?

astachowiczhabana commented:
@pytorchbot rebase

guangyey commented:
@pytorchbot merge

cyyever commented:
@pytorchbot merge -i

guangyey commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -18290,7 +18323,7 @@ def sample_inputs_alias_copy(op_info, device, dtype, requires_grad, **kwargs):
                        # Greatest absolute difference: nan at index (700,) (up to 0.01 allowed)
                        # Greatest relative difference: nan at index (700,) (up to 0.001 allowed)
-                       DecorateInfo(unittest.expectedFailure, 'TestUnaryUfuncs', 'test_reference_numerics_large',
+                       DecorateInfo(unittest.skip("Skipped!"), 'TestUnaryUfuncs', 'test_reference_numerics_large',
```
Why have this change.

guangyey commented:
@pytorchbot rebase

astachowiczhabana committed:
Skip failing tests on xpu with complex dtype on windows

On xpu we use std:: implementation of trig kernels. Issue comes from
differences in implementation of trigonometry functions on complex dtypes
in compiler headers. Windows compiler implementation is not conformant
with ISO 9899. These tests use numpy as reference and numpy is implemented
according to ISO 9899, hence those tests fail on Windows. Same failures
can be observed on cpu, and those tests are skipped there.

Fixes https://github.com/intel/torch-xpu-ops/issues/1195

albanD reviewed (approved):
Sounds good
